code/game/objects/items/holy_weapons.dm

```diff
@@ -1,3 +1,177 @@
+// CHAPLAIN CUSTOM ARMORS //
+
+/obj/item/clothing/head/helmet/chaplain
+	name = "crusader helmet"
+	desc = "Deus Vult."
+	icon_state = "knight_templar"
+	item_state = "knight_templar"
+	armor = list("melee" = 41, "bullet" = 15, "laser" = 5,"energy" = 5, "bomb" = 5, "bio" = 2, "rad" = 0, "fire" = 0, "acid" = 50)
+	flags_inv = HIDEMASK|HIDEEARS|HIDEEYES|HIDEFACE|HIDEHAIR
+	flags_cover = HEADCOVERSEYES | HEADCOVERSMOUTH
+	strip_delay = 80
+	dog_fashion = null
+
+/obj/item/clothing/suit/armor/riot/chaplain
+	name = "crusader armour"
+	desc = "God wills it!"
+	icon_state = "knight_templar"
+	item_state = "knight_templar"
+
+/obj/item/holybeacon
+	name = "armaments beacon"
+	desc = "Contains a set of armaments for the chaplain."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "gangtool-red"
+	item_state = "radio"
+	var/used = FALSE
+
+/obj/item/holybeacon/attack_self(mob/user)
+	if(user.mind && (user.mind.isholy) && !used)
+		beacon_armor(user)
+	else
+		playsound(src, 'sound/machines/buzz-sigh.ogg', 60, 1)
+
+/obj/item/holybeacon/proc/beacon_armor(mob/M)
+	if(SSreligion.holy_armor_type)
+		return
+	var/obj/item/storage/box/holy/holy_armor_box
+	var/list/holy_armor_list = typesof(/obj/item/storage/box/holy)
+	var/list/display_names = list()
+	for(var/V in holy_armor_list)
+		var/atom/A = V
+		display_names += initial(A.name)
+
+	var/choice = input(M,"What holy armor kit would you like to order?","Holy Armor Theme") as null|anything in display_names
+	if(QDELETED(src) || !choice || M.stat || !in_range(M, src) || M.restrained() || !M.canmove || used)
+		return
+
+	var/index = display_names.Find(choice)
+	var/A = holy_armor_list[index]
```

Comment:
I'm not sure what you're trying to do here but this is why it's not spawning.

Your index is returning a name, and A thinks h_a_l is an assoc list (it is not).

Reply:
hmmm

Reply:
@Armhulen Haven't tested but what you want here is something like:

```dm
var/list/display_names = list()
for(var/V in typesof(/obj/item/storage/box/holy))
    var/atom/A = V
    display_names[initial(A.name)] = A
var/choice = input(...) as null|anything in display_names
if (...)
    return
var/A = display_names[choice]
SSreligion.holy_armor_type = A
var/holy_armor_box = new A
...
```

BTW it seems the `used` var is redundant with `SSreligion.holy_armor_type`.

Reply:
Thanks, SpaceManiac!